torch/csrc/autograd/python_variable.cpp

```diff
@@ -143,7 +143,24 @@ static PyObject *THPVariable_pynew(PyTypeObject *type, PyObject *args, PyObject
   END_HANDLE_TH_ERRORS
 }
 
-// Instantiates a subclass of torch.Tensor. Used by nn.Parameter()
+// Instantiates a subclass of self with the same data.
+static PyObject* THPVariable_as_subclass(THPVariable* self, PyObject* args, PyObject* kwargs) {
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({
+    "as_subclass(PyObject* cls)",
+  });
+  ParsedArgs<2> parsed_args{};
+  auto r = parser.parse(args, kwargs, parsed_args);
+  PyObject* cls = r.pyobject(0);
+  if (!PyType_Check(cls)) {
+    throw TypeError("cls must be a type (got %s)", Py_TYPE(cls)->tp_name);
+  }
+  auto data = as_variable_ref(self->cdata.alias());
+  data.unsafeGetTensorImpl()->set_allow_tensor_metadata_change(true);
```

Comment:
I don't think you need this line anymore, `alias()` should allow metadata change automatically. You can test for this by running `resize_` on the result.